llvm/test/tools/sycl-post-link/composite-spec-constant.ll

```diff
@@ -0,0 +1,91 @@
+; RUN: sycl-post-link -spec-const=rt --ir-output-only %s -S -o - \
+; RUN: | FileCheck %s --implicit-check-not __sycl_getCompositeSpecConstantValue
+;
+; This test is intended to check that sycl-post-link tool is capable of handling
+; composite specialization constants by lowering them into a set of SPIR-V
+; friendly IR operations representing those constants.
+;
+; CHECK: %[[#NS0:]] = call i32 @_Z20__spirv_SpecConstantii(i32 [[#ID:]], i32
```

Comment:
What is value of `ID`? Is it 0?

Reply:
It is not necessary `0`, because if we have a composite spec constant, then the first ID will be dedicated to the composite itself and only the second available ID will be dedicated to its first member.

I'm using [FileCheck numeric substitutions](https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-numeric-substitution-blocks) here because numbers are more or less consecutive

Reply:
> It is not necessary 0, because if we have a composite spec constant, then the first ID will be dedicated to the composite itself and only the second available ID will be dedicated to its first member.

I updated algorithm in 9a82d5320c6a9e228fc183f211980b8de5d58d37, so no IDs are being reserved for composites anymore, but it is still not necessary is zero in general, because it depends on the fact whether some other specialization constants were encountered earlier.